pkg/pool-weighted/test/managed/BasePool.test.ts

```diff
@@ -0,0 +1,830 @@
+    });
+  });
+
+  describe('protocol fees', () => {
+    let pool: Contract;
+
+    sharedBeforeEach(async () => {
+      pool = await deployBasePool();
+    });
+
+    it('skips zero value mints', async () => {
+      const tx = await pool.payProtocolFees(0);
+
+      expectEvent.notEmitted(await tx.wait(), 'Transfer');
+    });
+
+    it('shares protocol fees collector with the vault', async () => {
+      expect(await pool.getProtocolFeesCollector()).to.be.eq(await vault.getProtocolFeesCollector());
+    });
+
+    it('mints bpt to the protocol fee collector', async () => {
+      const feeCollector = await pool.getProtocolFeesCollector();
+
+      const balanceBefore = await pool.balanceOf(feeCollector);
+      await pool.payProtocolFees(fp(42));
+      const balanceAfter = await pool.balanceOf(feeCollector);
+
+      expect(balanceAfter.sub(balanceBefore)).to.equal(fp(42));
+    });
+  });
+
+  describe('pause', () => {
+    let pool: Contract;
+    const PAUSE_WINDOW_DURATION = MONTH * 3;
+    const BUFFER_PERIOD_DURATION = MONTH;
+
+    let sender: SignerWithAddress;
+
+    function itCanPause() {
+      it('can pause', async () => {
+        await pool.connect(sender).pause();
+
+        const { paused } = await pool.getPausedState();
+        expect(paused).to.be.true;
+      });
+
+      context('when paused', () => {
+        let poolId: string;
+        let initialBalances: BigNumber[];
+
+        sharedBeforeEach('deploy and initialize pool', async () => {
+          initialBalances = Array(tokens.length).fill(fp(1000));
+          poolId = await pool.getPoolId();
+
+          const request: JoinPoolRequest = {
+            assets: tokens.addresses,
+            maxAmountsIn: initialBalances,
+            userData: WeightedPoolEncoder.joinInit(initialBalances),
+            fromInternalBalance: false,
+          };
+
+          await tokens.mint({ to: poolOwner, amount: fp(1000 + random(1000)) });
+          await tokens.approve({ from: poolOwner, to: vault });
+
+          await vault.connect(poolOwner).joinPool(poolId, poolOwner.address, poolOwner.address, request);
+        });
+
+        sharedBeforeEach('pause pool', async () => {
+          await pool.connect(sender).pause();
+        });
+
+        it('swaps revert', async () => {
+          const singleSwap: SingleSwap = {
+            poolId,
+            kind: SwapKind.GivenIn,
+            assetIn: tokens.get(0).instance.address,
+            assetOut: tokens.get(1).instance.address,
+            amount: 1, // Needs to be > 0
+            userData: '0x',
+          };
+
+          const funds: FundManagement = {
+            sender: poolOwner.address,
+            recipient: poolOwner.address,
+            fromInternalBalance: false,
+            toInternalBalance: false,
+          };
+
+          // min amount: 0, deadline: max.
+          await expect(vault.connect(poolOwner).swap(singleSwap, funds, 0, MAX_UINT256)).to.be.revertedWith('PAUSED');
+        });
```

Comment:
There's two swap handlers, one for the GeneralPool specialization, and one for the other two. This is only testing the General handler (because the Pool is a General Pool).

Reply:
Good catch; added.